smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/RebalanceListeners.java

```diff
@@ -33,8 +33,9 @@ static ConsumerRebalanceListener createRebalanceListener(
                 @Override
                 public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
                     log.executingConsumerRevokedRebalanceListener(consumerGroup);
-                    // TODO Why don't we call the commit handler?
                     try {
+                        reactiveKafkaConsumer.removeFromQueueRecordsFromTopicPartitions(partitions);
+                        commitHandler.partitionsRevoked(partitions);
```

Comment:
// TODO Why don't we call the commit handler?

We definitely should. This looks like a mistake. It probably hasn't been an issue because if the user is supplying there own listener there's a high chance they are using the ignore strategy and dealing with offsets themselves, examples:

Storing offsets else where for exactly once delivery
Always replace from a given timestamp and never commits (kafka backed in memory caches)

Reply:
I totally agree with this comment. But should we remove records from the queue for revoked topic partitions for all commit strategies or only for throttled, which has a high risk of creating duplicates on rebalance? I am thinking that we risk dropping them if we remove records from the queue when the commit strategy is NONE and auto.commit is enabled.

Reply:
Under none and auto commit it will suffer the same issue. On rebalance the vanilla Kafka consumer will rewind to committed offset for any assigned topic partition and for those that are revoked it shouldn't process be processing. 

Per the SRRM documentation if the strategy is nome and auto commit Is enabled we don't guarantee at least once delivery. Per the Kafka documentation it's not possible to do so with s reactive framework.

Reply:
I personally have a use case were I set the strategy to none and auto.commit to false and control the starting point from a supplied rebalance listened. I definitely don't want to process another instance's record after a rebalance. 

Ultimately that's the issue. Continuing to process records from revoked partitions is "wrong".  Under version 2.x of SRRM this wouldn't happen because of how the vertx client is designed irrespective of settings (strategy and auto commit).

Reply:
Actually the documentation says that NONE with auto commit does guarantee at least once delivery under certain conditions

https://github.com/smallrye/smallrye-reactive-messaging/blob/main/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaIgnoreCommit.java#L13

But that's no longer true under 3.9.1 (even without this PR). WIth auto commit true the vanilla client will commit every 5 seconds on poll. Under 2.2 that was true because a poll would only happen on request. The documentation should be update to state that there is no guarantee under NONE.

Reply:
I am not that familiar with the 2.2 implementation but I think this guarantee wasn't there even between 2.2 and 3.9.1.

Either the consumer stops polling until processing all received records, and therefore risks not responding to the heartbeat (SMRM 2.2), or the consumer continues polling (paused or not) in a separate thread while processing records, but risks auto committing records that are not yet processed (SMRM 3.9.1 with ignore commit strategy and enable.auto.commit=true). 

This leads me to the conclusion that in order to guarantee at-least-once delivery with SMRM Kafka we need offset tracking like throttled commit strategy or some custom offset checkpointing. We need to fix the Javadoc you mentioned and other documentation citing this.

Reply:
As Ozan said, the 2.2 implementations had a critical issue when not consuming (fast) enough and where you could be kicked out because you didn't call poll. We have to call poll frequently (even when all partitions are paused) because of the heartbeat.

Reply:
Right. The 2.2 implementation would only call poll on downstream requests. So this was true:

> When `enable.auto.commit` is `true` this strategy provides at-least-once delivery **IF** the channel processes the message without performing any asynchronous processing.


Big IF. Frankly if the user really wants to guarantee at-least-once they should stick with the throttled implementation or roll there own via a rebalance listener. The kafka docs are pretty adamant that if you are processing requests reactively then you are on your own **wrt at-least-once** guarantees and should not set `enable.auto.commit` to true.

Reply:
Went ahead and added a commit updating the docs